ol/miner/src/block.rs

```diff
@@ -104,32 +105,87 @@ pub fn mine_and_submit(
         status_err!("Genesis block_0.json not found. Exiting.");
         std::process::exit(0);
     } else {
+        // the max block that has been succesfully submitted to client
+        let mut remote_height = current_block_number.unwrap();
+        // in the beginning, mining height is +1 of client block number
+        let mut mining_height = remote_height + 1; 
+
         // mine continuously from the last block in the file systems
-        let mut mining_height = current_block_number.unwrap() + 1;
         loop {
+            status_info!(
+                "Latest client block:",
+                format!("{}", remote_height)
+            );
+
             status_info!(format!("Block {}", mining_height), "Mining VDF Proof");
 
-            let block = mine_once(&config)?;
+            let block_result = mine_once(&config);
+            let block: Block;
+            match block_result {
+                Ok(blk) => block = blk,
+                Err(err) => {
+                    status_warn!("Mining block {} failed: {}", mining_height, err);
+                    continue
+                }
+            }
+
             status_info!(
                 "Proof mined:",
                 format!("block_{}.json created.", block.height.to_string())
             );
-
-            if let Some(ref _node) = config.profile.default_node {
-                match commit_proof_tx(&tx_params, block.preimage, block.proof, is_operator) {
-                    Ok(tx_view) => match eval_tx_status(tx_view) {
-                        Ok(_) => status_ok!("Success:", "Proof committed to chain"),
-                        Err(_) => status_err!("Miner transaction rejected"),
-                    },
-                    Err(err) => status_err!("Miner transaction rejected: {}", err),
+            
+            // submits backlog to client
```

Comment:
We already have a method called backlog::process_backlog(), is there a reason we are not using that here?

Reply:
The backlog method might need some modifications though. Will do and push.

Reply:
Closed